### PR TITLE
docs/libvirt: Add docs around resources

### PR DESCRIPTION
**Pull request by cgwalters**
8GB seems to be important now.  Also elaborate more on the rationale.

### CODE REVIEW

**Comment by praveenkumar** [comment]
Thanks for writing this up. I also think we should increase the default limit also from 4GB to 8GB since there is no way this is working at this moment and this options should be there to make sure if a user need more resource than default. Default should be always working (which is not currently).

File: docs/dev/libvirt-howto.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Libvirt HOWTO
 
-Launching clusters via libvirt is especially useful for operator development.
+The libvirt backend for the installer is maintained for development; it is not a production use case.
+
+It is particularly useful for debugging issues related to the operating system and bootstrapping, as one has much easier access to debugging information than in a cloud.
+You also avoid incurring costs for running in a cloud.
```
Comment (praveenkumar) [inline docs/dev/libvirt-howto.md]: 👍

File: docs/dev/libvirt-howto.md [hunk 2]
```diff
@@ -203,15 +206,33 @@ This step allows installer and users to resolve cluster-internal hostnames from
...
 
+## Running the installer
+
+Now, most of the [the usual quick-start](../../README.md#quick-start) applies.
+However, the current amount of RAM and CPU allocated to the master is commonly too small for a reliable installation.
+You can configure this via environment variables.  At the time of this writing, at least 8GB is recommended
+for the master:
+
+```sh
+env TF_VAR_libvirt_master_memory=8192 ./bin/openshift-install ...
```
Comment (praveenkumar) [inline docs/dev/libvirt-howto.md]: 👍

**Comment by zeenix** [comment]
/lgtm

@cgwalters Could you please rebase this so we can merge it?

**Comment by cgwalters** [comment]
Rebased :surfing_man:

**Comment by zeenix** [comment]
@cgwalters 👍

/lgtm

**Comment by zeenix** [comment]
Seems being an "approver" doesn't give me merge powers. :( @abhinavdahiya could you kindly do the honours? Thanks.

**Comment by praveenkumar** [comment]
@zeenix `ci/prow/e2e-aws-upgrade` job is still in pending which is `Required` once this is executed successfully it will be auto merge since the label already added by CI.

**Comment by zeenix** [comment]
> @zeenix `ci/prow/e2e-aws-upgrade` job is still in pending which is `Required` once this is executed successfully it will be auto merge since the label already added by CI.

Ah ok. I blame GH for not keeping the CI jobs view up2date so it showed me all is done accept merging.

**Comment by abhinavdahiya** [comment]
/hold

If we are documenting to override 8GB 4cores for control plane, users question would be why is that not the default?

**Comment by praveenkumar** [comment]
> If we are documenting to override 8GB 4cores for control plane, users question would be why is that not the default?

@abhinavdahiya I do think that should be default I think I once asked also same.

**Comment by zeenix** [comment]
> @zeenix Does the default installation currently OOM when using 4GiB?

You need at least 6GiB currently for each node. Not sure about OOM but Prometheus components won't start if they see less than 6GiB on a node.

> If so I think we should definitely change that to whatever is required to succeed all of the time wrt to memory.

The default resources have been bumped already a while ago. I think this issue is stale now so closing. @cgwalters please feel free to re-open if you want to update the patch. Thanks.

/close

**Pull request closed without merging**